### ISSUE TITLE
[BUG] NAM GUI not picking best ESR

### ISSUE THREAD

**Issue by Geeztown** [sdatkinson/neural-amp-modeler#247]
When using customized settings in core.py and a custom test tone to train a difficult to train fuzz pedal, I'm finding that the ESR of the final NAM file that gets saved and it's corresponding plot are not the lowest ESR saved under checkpoints.

**To Reproduce**
1. Download Custom test tone, core.py file, and HP-1 5.wav as the output file from https://drive.google.com/drive/folders/1-8L01n2AvVHaEbSvooUjTumyj-VXeBBk?usp=sharing
2. Replace Core.py in nam/train with the downloaded core.py file.
3. You man need to change num workers value in core.py depending on your machine. I have it set to 12. default is 0.
4. Open Anaconda Prompt, activate and run nam.
5. Select the custom test tone for the input, HP-1 5.wav as the output.  Store finished nam file wherever you choose.
6. click advanced and change Epochs to 30.
7. Run it.
8. Look under the lightning logs / version / checkpoints folder and compare ESR scores with the final score from Running nam.

**Expected behavior**
The finished .nam file and corresponding plot png should show the lowest ESR score that you see under checkpoints.  But it doesn't do that.

**Screenshots**
![HP-1 5](https://github.com/sdatkinson/neural-amp-modeler/assets/19899319/ecdb1bd7-dbed-42fb-a760-9bcab7135274)
![Lowest ESR](https://github.com/sdatkinson/neural-amp-modeler/assets/19899319/edd6d72e-17a0-4f22-b83d-5dc101ddcb7e)

**Desktop (please complete the following information):**
 - OS Windows 11 Pro 22H2 22621.1702
 - Running locally. 
 - NAM version 0.5.3

**Additional context**
Changes I made to Core.py

- delay to 0 
- Dilations value to 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512
- num workers to 12
- lr to 0.001
- lr decay to 0.001
- input_version: Optional[Version] = Version(1, 1, 1),
- save_top_k from 3 to 10
- added ESR to current checkpoint  - pl.callbacks.model_checkpoint.ModelCheckpoint(
                filename="checkpoint_last_{epoch:04d}_{step}_{ESR:.4f}_{MSE:.3e}", every_n_epochs=1

I tried using the same core.py with the standard versoin 1_1_1 test tone file and a less distorted less complex output file from the same fuzz pedal and didn't have this issue.  However with 100 Epochs, my best ESR was 0.1192.  It took less time with shorter / smaller wav files, but wasn't as accurate even with more epochs.  It's possible my custom test tone file is causing the issue.  It's also possible that more data, and processing a more complex output could be exposing the issue.

Thank you for looking into this, and thank you for all your hard work on this project!  It's really inspiring, I've learned a lot about machine learning because of you, and you've made it possible for me to model a rare fuzz pedal that my grandpa, who passed away in 1993, built.  And that's really important to me on a personal level.  God bless you sir!

**Comment by Geeztown** [comment]
I just looked up some info on padding in a neural network.  It sounds like padding is added to preserve the data at the edges of the matrix which should give you more accuracy.  So you’re saying that the validation during training is losing some data and not quite as accurate as the final plot which is using padding as specified in the model?

If that’s true, it could be picking the best check point, but when it uses padding, the ESR score is higher?


To clarify, if I use the last 10 seconds of the original test tone file, I’m getting a little bit of that wobbly guitar before it’s silent.  Would it be better to just add a second of silence before the last 9 seconds validation?


I agree that ESR doesn’t tell the full story.  But I’ve found that better ESRs are giving me better results when doing a null test.  To be clear, the info I shared above was a less gpu intensive version of what I've been doing to reproduce the bug.  To get good results with this, I'm using even more dilations, and running 500 epochs.  But the results are worth it.  After 13 hours, I'm getting an ESR of 0.0010 in the checkpoints and about -30db in the null test.  I was getting about -17 or so and some transient spikes before.  Plus, what's left over seems to be mostly white noise now.

But I also have seen people with higher ESRs do better than I am on a null test.  And a null test doesn’t necessarily tell the full story either.  What really matters is how it sounds and feels when you play through it.

**Comment by Geeztown** [comment]
I tried adding 1 second of silence before the last 9 seconds of the original file.  That seems to have fixed the issue.  And you were right, I'm getting a little better ESR scores as well.

I'm not totally sure what was causing that low frequency ringing to begin with.  I still had it after taking the pedal out of the loop.  So either my Little Labs Redeye DI/Reamp boxes, or my apollo x4.  The pedal seemed to amplify it, so I would guess it's my little labs redeye reamp box.  I know EQ can cause phase shift, and at first I thought maybe it was the antialiasing filter in my converters, but it seems like kind of a lot of ringing for it to be eq related.  

For now, I'm happy with the models I'm getting, and I'm glad it wasn't a bigger issue.

Thanks again for you help!

**Comment by sdatkinson** [comment]
Great! Going to close this and we can work on a feature that makes sure others don't make this mistake in #252.
